yare-core/src/main/java/com/sabre/oss/yare/core/EngineControllerObservable.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * MIT License
+ *
+ * Copyright 2018 Sabre GLBL Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sabre.oss.yare.core;
+
+public interface EngineControllerObservable extends EngineController {
```

Comment:
I would get rid of this interface - one implementation is enough (`DefaultEngineController`).

Reply:
This implementation is weak from my side and name for this interface would be `EngineControllerListenable` and it shouldn't extend `EngineController`, but my idea was that client from action shouldn't have access to `register(Listener)` method and engine shouldn't have access to `stopProcessing()` method, so I've added `EngineControllerObservable` to this purpose. If we remove this interface `RulesEngineBuilder` will depend on specific implementation (`DefaultEngineController`). What about `DefaultEngineController` implementing `EngineControllerListenable` and `EngineContoller`?

Reply:
This is exactly what I expect - `DefaultEngineController` is a pass-thru object only. I don't expect any other implementation so it is ok to use it in `RulesEngineBuilder`.